nanoFramework.Tools.DebugLibrary.Shared/WireProtocol/Commands.cs

```diff
@@ -528,6 +551,103 @@ public override bool PrepareForSend(byte[] data, int length, int offset = 0)
                 return true;
             }
         }
+       
+        /// <summary>
+        /// Perform storage operation on the target device.
+        /// </summary>
+        public class Monitor_StorageOperation : OverheadBase
+        {
+            public uint Operation = (byte)StorageOperation.None;
+            public uint NameLength = 0;
+            public uint DataLength = 0;
+            public byte[] Data = new byte[0];
+
+            public class Reply
+            {
+                public uint ErrorCode;
+            };
+
+            public void PrepareForSend(
+              StorageOperation operation,
+              string name)
+            {
+                Operation = (byte)operation;
+                Data = Encoding.UTF8.GetBytes(name);
+                NameLength = (uint)Data.Length;
+            }
+
+            public void PrepareForSend(
+                StorageOperation operation,
+                string name,
+                byte[] data,
+                int offset,
+                int length)
+            {
+                Operation = (byte)operation;
+                Data = Encoding.UTF8.GetBytes(name);
+                NameLength = (uint)(Data.Length + data.Length);
+
+                PrepareForSend(data, offset, length);
+            }
+
+            public override bool PrepareForSend(
+                byte[] data,
+                int offset,
+                int length)
+            {
+                DataLength = (uint)length;
```

Comment:
I'm finding this logic a bit confusing...
We always have to send the file name, correct?
If we get to this `PrepareForSend` method its because we are sending data, therefore we always to:
1. resize the `Data` array to accommodate the data coming from the parameter
2. copy the data from to the resized `Data` array